Text/Pandoc/Builder.hs

```diff
@@ -483,9 +493,7 @@ table caption cellspecs headers rows = singleton $
          numcols  = case (headers:rows) of
                          [] -> 0
                          xs -> maximum (map length xs)
-         headers' = if null headers
-                       then replicate numcols mempty
-                       else headers
+         headers' = pad mempty numcols headers
```

Comment:
Since the `pad` function is only used here, I would move it here: on the same level as `headers'` and `numcols`.